src/Shouldly.Tests/ConventionTests/ApprovePublicApi.ShouldlyApi.approved.cs

```diff
@@ -330,6 +351,47 @@ public class ShouldContainWithCountShouldlyMessage : Shouldly.ShouldlyMessage
     {
         public ShouldContainWithCountShouldlyMessage(object? expected, object? actual, int matchCount, string? customMessage, [System.Runtime.CompilerServices.CallerMemberName] string shouldlyMethod = null) { }
     }
+    public class ShouldMatchApprovedException : Shouldly.ShouldAssertException
+    {
+        public ShouldMatchApprovedException(string? message, string? receivedFile, string? approvedFile) { }
+    }
+    [Shouldly.ShouldlyMethods]
+    public static class ShouldMatchApprovedTestExtensions
+    {
+        public static void ShouldMatchApproved(this string actual, System.Action<Shouldly.ShouldMatchConfigurationBuilder>? configureOptions = null, string? customMessage = null) { }
+    }
+    public class ShouldMatchConfiguration
+    {
+        public ShouldMatchConfiguration() { }
+        public ShouldMatchConfiguration(Shouldly.ShouldMatchConfiguration initialConfig) { }
+        public string? ApprovalFileSubFolder { get; set; }
+        public Shouldly.IDiffEngine? DiffEngine { get; set; }
```

Comment:
new

Reply:
I really like this overall approach and think it's the way to go.

My only thought is calling it `IDiffEngine` feels accidentally coupled to the implication.
Can I suggest either:
* `IDiffViewer`
* `IDiffToolLauncher`
* ... something else, I'm welcome to suggestions

Reply:
I agree.  I also like `IDiffViewer` but whatever you think is fine.  Should I try that?  Let me know and I'll change the relevant property, interface, package name, folder, etc.

Reply:
Actually I would probably leave the package name alone, since it is tied to DiffEngine

Reply:
Yeah, keep `Shouldly.DiffEngine` 👍 

Let's go with `IDiffViewer` and `DiffViewer` for the config property 🙂

Reply:
Done